### ISSUE TITLE
Pass in configuration using environment variable

### ISSUE THREAD

**Issue by xaka** [elastic/logstash#4376]
As we live in era where containers taking over place, it would be nice to assign configuration to environment variable, may be `$LOGSTASH`, and make sure logstash consumes it when no configuration file (`-f`) or configuration string (`-e`) is provided. The reason is pretty simple - I don't want to build my own image from base one just to inject some file nor I want to mess around with volumes nor I'm happy with `-e` because you end up with a long unreadable line and at the same, when you use docker compose for instance, you can come up with (take a look at magical pipe `|` supported by YAML):

```
image: logstash
environment:
  LOGSTASH: |
    input {
      ...
    }
```

Which is pretty readable and allows one to use existing official image.

**Comment by xaka** [comment]
For now I'm using the following workaround to stick to official image without need for building my own:

```
image: logstash
command: bash -c 'echo $$LOGSTASH > logstash.conf; exec logstash -f logstash.conf'
environment:
  LOGSTASH: |
    input {
      ...
    }
```

**Comment by suyograo** [comment]
Fixed in https://github.com/elastic/logstash/issues/3944, will be available in 2.3
